### PR TITLE
`Offering`: new `getMetadataValue` with `Decodable` type

### PR DESCRIPTION
**Pull request by NachoSoto**
This allows to easily extract data from `Offering.metadata` in a structured way.

Example:
```swift
struct Data: Decodable {
    var number: Int
}

let elements: [Data] = offering.getMetadataValue(for: "elements") ?? []
let element: Data? = offering.getMetadataValue(for: "element")
```

### CODE REVIEW

File: Sources/Purchasing/Offering.swift [hunk 1]
```diff
@@ -215,17 +215,37 @@ import Foundation
...
             return `default`
         }
         return value
     }
 
+    ///
+    /// - Returns: the `metadata` value associated to `key` for the expected `Decodable` type,
+    /// or `nil` if not found.
+    /// - Throws: Error if the content couldn't be deserialized to the expected type.
+    ///
```
Comment (joshdholtz) [inline Sources/Purchasing/Offering.swift:233]: Should we mention which `KeyDecodingStrategy` its using?

**Review by joshdholtz — COMMENTED** [review]
This looks amazing! Just one quick question about `KeyDecodingStrategy`

File: Sources/Purchasing/Offering.swift (hunk 1 above)
Comment (NachoSoto) [reply Sources/Purchasing/Offering.swift]: Done.

**Review by joshdholtz — APPROVED** [review]
:shipit:

**Pull request merged**